iceoryx_utils/source/posix_wrapper/unix_domain_socket.cpp

```diff
@@ -87,17 +84,36 @@ UnixDomainSocket& UnixDomainSocket::operator=(UnixDomainSocket&& other)
         m_sockfd = std::move(other.m_sockfd);
         m_sockAddr = std::move(other.m_sockAddr);
         other.m_sockfd = INVALID_FD;
+        m_maxMessageSize = std::move(other.m_maxMessageSize);
     }
 
     return *this;
 }
 
-cxx::expected<bool, IpcChannelError> UnixDomainSocket::exists(const std::string& name)
+cxx::expected<bool, IpcChannelError> UnixDomainSocket::unlinkIfExists(const std::string& name) noexcept
 {
-    return cxx::success<bool>(true);
+    if (name.empty() || name.size() < SHORTEST_VALID_NAME || name.size() > LONGEST_VALID_NAME || name.at(0) != '/')
+    {
+        return cxx::error<IpcChannelError>(IpcChannelError::INVALID_CHANNEL_NAME);
+    }
+    auto nameCStr = name.c_str();
+    nameCStr++; // don't use the leading '/' for the unix domain socket name
+
+    auto unlinkCall =
+        cxx::makeSmartC(unlink, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {ERROR_CODE}, {ENOENT}, nameCStr);
+
+    if (!unlinkCall.hasErrors())
+    {
+        // ENOENT is set if this socket is not known
```

Comment:
@note here

Reply:
Developer comment. Not intended for the documentation.